### ISSUE TITLE
Reactive queries sometimes return a corrupt DB error

### ISSUE THREAD

**Issue by ospfranco** [OP-Engineering/op-sqlite#128]
**Describe the bug**
[Reproducible example](https://github.com/maksym-kyryliuk/opsqliteIssue).

It seems the problem is executing the query again in the middle of a transactions with deletes or indexes or foreign keys?. The update hook fires in between each of the statements in a executeBatch (which are wrapped in a transaction) and sqlite just kaputs itself, because:

> The update hook implementation must not do anything that will modify the database connection that invoked the update hook. Any actions to modify the database connection must be deferred until after the completion of the sqlite3_step() call that triggered the update hook. Note that sqlite3_prepare_v2() and sqlite3_step() both modify their database connections for the meaning of "modify" in this paragraph.

I've tried adding a mutex but to no avail since everything runs on the same thread and just skipping until the batch operation has finished is also no good since only updates trigger the reactive query. This means a different implementation of the reactive queries is necessary. Right now the update hook triggers the re-execution of the prepared query internally, which will break the rule above. We need to queue the statements and re-execute them until it is safe to do so, so a major re-engineering effort is necessary.

**Comment by ospfranco** [comment]
no, no workaround at the moment. I need to re-write how the callbacks are executed, I just don't have the time for it.

**Comment by rvibit** [comment]
> no, no workaround at the moment. I need to re-write how the callbacks are executed, I just don't have the time for it.

I was just trying different ways and added this hacky code for now and working fine for my use case.

```js
db.reactiveExecute({
  query: query, //a query with left join
  arguments: [],
  fireOn: [
    {
      table: TableName.ACCOUNTS,
    },
    {
      table: TableName.TRANSACTIONS,
    },
  ],
  callback: (res) => {
    try {
//res is currpt DB here in delete case so added a condition to execute the query manually in else block
      const accounts = res?.rows?._array;
      if (accounts) {
        defaultStore.set(accountsAtom, accounts);
      } else {
        const data = db.execute(query).rows?._array;
        defaultStore.set(accountsAtom, data);
      }
    } catch (error) {
      console.log(error);
    }
  },
});

**Comment by devYonz** [comment]
I am also experiencing crashes, but the try catch did not fix my problem.

**Comment by rvibit** [comment]
> I am also experiencing crashes, but the try catch did not fix my problem.

Try catch is not solving the problem , the if else condition is the key, when the crash happenes accounts is undefined so wrapping it in if worked for me

**Comment by devYonz** [comment]
My situation is that the whole app crashes, next I kept needing to have executed an regular query before being able to use reactive query. We finally have a stable reactive query after we turned off the thread unsafe perf mode and started using the cr-sqlite binary.  In other wods, set your package.json to: 
```
  "op-sqlite": {
    "sqlcipher": false,
    "crsqlite": true,  <--- 
    "performanceMode": "2", <----
    "iosSqlite": false,
    "sqliteFlags": "-DSQLITE_DQS=0",
    "fts5": false,
    "libsql": false,
    "sqliteVec": true
  }
}
```
